src/build.rs

```diff
@@ -86,7 +87,71 @@ fn patch_lib_kind_in_target(ws: &mut Workspace, libkinds: &[&str]) -> anyhow::Re
     Ok(())
 }
 
-/// Build import library for Windows
+/// Build def file for windows-msvc
+fn build_def_file(
+    ws: &Workspace,
+    name: &str,
+    target: &target::Target,
+    targetdir: &PathBuf,
+) -> anyhow::Result<()> {
+    let os = &target.os;
+    let env = &target.env;
+
+    if os == "windows" && env == "msvc" {
+        ws.config()
+            .shell()
+            .status("Building", ".def file using dumpbin")?;
+
+        let txt_path = targetdir.join(format!("{}.txt", name));
+        let mut dumpbin = std::process::Command::new("dumpbin");
+        dumpbin
+            .arg("/EXPORTS")
+            .arg(targetdir.join(format!("{}.dll", name)));
+        dumpbin.arg(format!("/OUT:{}", txt_path.to_str().unwrap()));
+
+        let out = dumpbin.output()?;
+        if out.status.success() {
+            let txt_file = File::open(txt_path)?;
+            let buf_reader = BufReader::new(txt_file);
+            let mut def_file = File::create(targetdir.join(format!("{}.def", name)))?;
+            writeln!(def_file, "{}", "EXPORTS".to_string())?;
+
+            // The Rust loop below is analogue to the following loop.
+            // for /f "skip=19 tokens=4" %A in (file.txt) do echo %A > file.def
+            // The most recent versions of dumpbin adds three lines of copyright
+            // information before the relevant content.
+            // If the "/OUT:file.txt" dumpbin's option is used, the three
+            // copyright lines are added to the shell, so the txt file
+            // contains three lines less.
+            // The Rust loop first skips 16 lines and then, for each line,
+            // deletes all the characters up to the fourth space included
+            // (skip=16 tokens=4)
```

Comment:
Thanks.

I just wanted to know if the output syntax/text is stable between versions, i.e. if a regex-based solution is more "safe" or not. If you're satisfied this approach is stable enough, I have no objections.

Reply:
Sincerely I've thought of a regex-based approach, but it would have complicated more the code in my opinion. So, after having looked around a bit, I've discovered that the current approach is the most used for `.def` files and that's why I've decided to use that, in this way we are more compliant with the most used implementations.